### ISSUE TITLE
Compiled-in environment/arg limits causing issues

### ISSUE THREAD

**Issue by hlein** [netblue30/firejail#3851]
**Bug and expected behavior**
https://github.com/netblue30/firejail/commit/88258569cb5f455dd39447867e559bfba20d91c7 added some reasonable-looking limits on environment (number and size of environment variables) and arguments (number, maybe also size?).

Except... this causes some regressions with things that used to work fine.

It would be great if these limits could be specified in firejail config files rather than compiled in, so that I could then arrange for .local overrides where needed. Or failing that, activated by an on-by-default flag like `--envcheck`, which could be suppressed with `--ignore-envcheck`.

Examples:

On gentoo, during package building, some keywords and targets get expanded into quite-large environment variables. For instance the build system keeps track of which Python versions you have installed, and which targets / USE flags (optional compile-time features) you have enabled; some package-flag / dependency calculations then explode if you have say python-2.7, python-3.6, python-3.7, and python-3.8 all installed at once. This can exceed the current hardcoded limits, so builds can fail at strange points (patch, strings, fontforge for some reason) if the build user's processes are being firejail'ed. If I could arrange for the build user to have higher limits, it could still benefit from firejail otherwise, and the rest of the system's firejail uses could benefit from the reasonable lower defaults.

I also have some file-syncing scripts / Makefiles that bundle up different file sets and rsync them by explicit argument list (`rsync a b c d e f g h i j ... targethost:`). I could modify these to pipe a list to rsync and do `--files=from=-`, but it would be simpler to just increase the ceilings in an `rsync.local` and `ssh.local`, and/or pass a `--ignore-envcheck` to firejail in those cases.

**Comment by SkewedZeppelin** [comment]
Yep.
I've had these patched out in my personal builds forever.

**Comment by rusty-snake** [comment]
Fixed by #3322?

**Comment by rusty-snake** [comment]
Duplicate of #3678?

**Comment by hlein** [comment]
> Fixed by #3322?

Nope. Using b5a40359125db3a2df9a6f6e9c12c476dd9b150b, which should include 1c7ea15b30d49d32a0e3cb79152514f1aeb19397 from #3322:

In the "rsync a ridiculously long list of filenames" test, I get:

`Error: too many arguments`

Same thing when emerging a package on Gentoo when `patch` has been firejail'ed.

**Comment by hlein** [comment]
> Duplicate of #3678?

Hm, related but I don't think a dupe. #3678 seems to be strictly about "a single env var that exceeds a length limit", but in this case both the number of env vars, and their lengths, cause problems.

**Comment by kmk3** [comment]
> > Duplicate of [#3678](https://github.com/netblue30/firejail/issues/3678)?
> 
> Hm, related but I don't think a dupe.
> [#3678](https://github.com/netblue30/firejail/issues/3678) seems to be
> strictly about "a single env var that exceeds a length limit", but in this
> case the number of arguments, the number of env vars, and the env var lengths
> cause problems.

Closing as a duplicate of:

* #3678
* #4633

As making the count/length configurable for both arguments and environment
variables should also resolve this issue.

If that is not the case, please clarify.
